src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -618,445 +501,1122 @@ private static PackOptions GetPackOptions(PackageSpec packageSpec, JObject rawPa
+                    case "centralPackageVersionsManagementEnabled":
+                        centralPackageVersionsManagementEnabled = ReadNextTokenAsBoolOrFalse(jsonReader, packageSpec.FilePath);
+                        break;
+
+                    case "configFilePaths":
+                        configFilePaths = jsonReader.ReadStringArrayAsList();
+                        break;
+
+                    case "crossTargeting":
+                        crossTargeting = ReadNextTokenAsBoolOrFalse(jsonReader, packageSpec.FilePath);
+                        break;
+
+                    case "fallbackFolders":
+                        fallbackFolders = jsonReader.ReadStringArrayAsList();
+                        break;
+
+                    case "files":
+                        jsonReader.ReadObject(filePropertyName =>
+                        {
+                            files = files ?? new List<ProjectRestoreMetadataFile>();
+
+                            files.Add(new ProjectRestoreMetadataFile(filePropertyName, jsonReader.ReadNextTokenAsString()));
+                        });
+                        break;
+
+                    case "frameworks":
+                        targetFrameworks = ReadTargetFrameworks(jsonReader);
+                        break;
+
+                    case "legacyPackagesDirectory":
+                        legacyPackagesDirectory = ReadNextTokenAsBoolOrFalse(jsonReader, packageSpec.FilePath);
+                        break;
+
+                    case "originalTargetFrameworks":
+                        originalTargetFrameworks = jsonReader.ReadStringArrayAsList();
+                        break;
+
+                    case "outputPath":
+                        outputPath = jsonReader.ReadNextTokenAsString();
+                        break;
+
+                    case "packagesConfigPath":
+                        packagesConfigPath = jsonReader.ReadNextTokenAsString();
+                        break;
+
+                    case "packagesPath":
+                        packagesPath = jsonReader.ReadNextTokenAsString();
+                        break;
+
+                    case "projectJsonPath":
+                        projectJsonPath = jsonReader.ReadNextTokenAsString();
+                        break;
+
+                    case "projectName":
+                        projectName = jsonReader.ReadNextTokenAsString();
+                        break;
+
+                    case "projectPath":
+                        projectPath = jsonReader.ReadNextTokenAsString();
+                        break;
+
+                    case "projectStyle":
+                        string projectStyleString = jsonReader.ReadNextTokenAsString();
+
+                        if (!string.IsNullOrEmpty(projectStyleString)
+                            && Enum.TryParse<ProjectStyle>(projectStyleString, ignoreCase: true, result: out var projectStyleValue))
```

Comment:
nit: don't think you need the var here...is there a reason you include?

Reply:
I think he wants to preserve `projectStyle` in case the Parsing fails. If he doesn't create a new variable, then it'll overwrite it with null upon failure.

Reply:
@donnie-msft is correct.

Like it or not, that was the [existing behavior](https://github.com/NuGet/NuGet.Client/blob/8a1d1c01bc27c282d20668dbbe1abd2182f9b3fb/src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs#L208-L213).  I only preserve it.